app/models/user.rb

```diff
@@ -69,14 +71,15 @@ def self.upsert_from_auth_hash(auth_hash)
     user.update_from_sso(firstname: auth_hash[:info][:first_name],
                          lastname: auth_hash[:info][:last_name],
                          email: auth_hash[:info][:email],
-                         room: auth_hash[:extra][:raw_info][:room])
+                         room: auth_hash[:extra][:raw_info][:room],
+                         username: auth_hash[:extra][:raw_info][:preferred_username])
```

Comment:
I just realized, to support this new field in the `Local login (no SSO)` flow, you need to add a field for it in [`config/initializers/omniauth.rb`](https://github.com/rezoleo/lea5/blob/master/config/initializers/omniauth.rb)

Reply:
You also need to add it to [`db/seeds.rb`](https://github.com/rezoleo/lea5/blob/master/db/seeds.rb)